### ISSUE TITLE
ValidatorTextField.shouldAllowViolation is behaving unexpectedly 

### ISSUE THREAD

**Issue by sparrowlegs** [ustwo/formvalidator-swift#43]
### Description

The `ValidatorTextField` incorrectly 'eats' characters when the `shouldAllowViolation` property is set false `false`.

### Steps to Reproduce

1. In the iOS Example project.  Change the `nameEntry` property in the `FormView` class to allow violations:
```swift
        nameEntry.textLabel.text = NSLocalizedString("Surname", comment: "")
        nameEntry.textField.shouldAllowViolation = true
        stackView.addArrangedSubview(nameEntry)
```
 
2. Run the app and type the letter 'A' in the Surname field.  So far, so good.
3. Finally, type the number '1' in the Surname field.  

### Expected Result

The Surname field should contain the text  'A1' and be highlighted in red with the error text 'Enter letters only'.

### Actual Result

The Surname field contains only text 'A' and no error state is indicated.

### Version Information

The current version I am using is:

- Xcode: 8.2.1 (8C1002)
- iOS/tvOS: iPhone 7 simulator running iOS 10.2
- FormValidator: 1.2.0

### Other Information

I'm not sure if the following should be separate bugs.  If you like, I can create them.

With the `shouldAllowViolation` property set to true, an empty Surname field will allow the user to type one number.  This shows the expected error state, but does not allow subsequent numbers to be typed.

Also, with the single number in the Surname field, deleting this character does not return the field back to it's non-error state.

**Comment by ghost** [comment]
@sparrowlegs Thanks for reporting this! We'd welcome a PR if you'd like to take a stab at fixing it.

**Comment by sparrowlegs** [comment]
@aamctustwo I'll give it a stab.

The `shouldAllowViolation` property is defined on `ValidatorControl` and `Condition`.  Which one is meant to have precedence?

If you look at line 133 of `ValidatorTextField.swift`, you'll see why I'm asking:
```swift
        if !validatorTextField.validateOnFocusLossOnly && range.location != 0,
            let conditions = conditions,
            (!validatorTextField.shouldAllowViolation || conditions[0].shouldAllowViolation) {
                return false
        }
```

**Comment by ghost** [comment]
Closed by #44 `Feature/issue 43 fix should allow violation`.
